### PR TITLE
CODEOWNER update for obs-infraobs-integrations.

### PR DESCRIPTION
**Pull request by lalit-satapathy**
As part of the _Beats repository quality gates effort_, need to keep the CODEOWNERS for the beats repo up-to-date.

Updating the beats modules/inputs which were owned by @elastic/obs-infraobs-integrations
- Left the entries which are either ambiguous (example: module/icinga)  not owned(example: module/kibana). 
- Added modules not listed earlier (example: many in /metricbeat/module/*)
- Trying to keep it consistent with [CODEOWNERS](https://github.com/elastic/integrations/blob/main/.github/CODEOWNERS) in the integration repo.

There will be still modules left in @elastic/integrations which needs to be cleaned-up.

Please review and let's know, if anything wrongly mapped.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -24,25 +24,25 @@ CHANGELOG*
...
-/filebeat/module/kafka @elastic/integrations
+/filebeat/module/iis @elastic/obs-infraobs-integrations
+/filebeat/module/kafka @elastic/obs-infraobs-integrations
 /filebeat/module/kibana @elastic/integrations
 /filebeat/module/kibana/ @elastic/infra-monitoring-ui
 /filebeat/module/logstash @elastic/integrations
 /filebeat/module/logstash/ @elastic/infra-monitoring-ui
-/filebeat/module/mongodb @elastic/integrations
+/filebeat/module/mongodb @elastic/obs-infraobs-integrations
 /filebeat/module/mysql @elastic/security-external-integrations
```
Comment (muthu-mps) [inline .github/CODEOWNERS:39]: Does this falls under obs-infraobs-integrations ?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -24,25 +24,25 @@ CHANGELOG*
...
+/filebeat/module/nginx @elastic/obs-infraobs-integrations
 /filebeat/module/osquery @elastic/security-asset-management
 /filebeat/module/pensando @elastic/security-external-integrations
-/filebeat/module/postgresql @elastic/integrations
-/filebeat/module/redis @elastic/integrations
+/filebeat/module/postgresql @elastic/obs-infraobs-integrations
+/filebeat/module/redis @elastic/obs-infraobs-integrations
 /filebeat/module/santa @elastic/security-external-integrations
 /filebeat/module/system @elastic/elastic-agent-data-plane
 /filebeat/module/traefik @elastic/integrations
```
Comment (muthu-mps) [inline .github/CODEOWNERS:48]: The `traefik` integration is currently owned by `obs-infraobs-integrations.`

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -133,19 +152,31 @@ CHANGELOG*
...
 /x-pack/filebeat/module/zoom @elastic/security-external-integrations
 /x-pack/filebeat/module/zscaler @elastic/security-external-integrations
 /x-pack/filebeat/processors/decode_cef/ @elastic/security-external-integrations
 /x-pack/heartbeat/ @elastic/uptime
 /x-pack/metricbeat/ @elastic/elastic-agent-data-plane
 /x-pack/metricbeat/docs/ # Listed without an owner to avoid maintaining doc ownership for each input and module.
 /x-pack/metricbeat/module/ @elastic/integrations
+/x-pack/metricbeat/module/activemq @elastic/obs-infraobs-integrations
+/x-pack/metricbeat/module/airflow @elastic/obs-infraobs-integrations
+/x-pack/metricbeat/module/cloudfoundry @elastic/obs-infraobs-integrations
```
Comment (kaiyan-sheng) [inline .github/CODEOWNERS:165]: how about `/x-pack/metricbeat/module/aws`, `/x-pack/metricbeat/module/awsfargate`, `/x-pack/metricbeat/module/azure` and `/x-pack/metricbeat/module/gcp`?

Comment (lalit-satapathy) [reply .github/CODEOWNERS:165]: Not covering cloud packages in this PR currently. @tommyers-elastic, Can we create separate PR for for the cloud packages?

**Comment by ishleenk17** [comment]
@lalit-satapathy : Ownership of etcd lies with IM-Integrations. 
We should update the same for module as well

**Comment by lalit-satapathy** [comment]
Opening for additional feedbacks if any.

CC: @joshdover @ruflin @andresrc

File: .github/CODEOWNERS (hunk 1 above)
Comment (lalit-satapathy) [reply .github/CODEOWNERS:39]: Looks like regular o11y [logs](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-mongodb.html). Where it should go?

File: .github/CODEOWNERS (hunk 2 above)
Comment (lalit-satapathy) [reply .github/CODEOWNERS:48]: Not sure where exactly [traefik](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-traefik.html) belong?

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -24,25 +24,25 @@ CHANGELOG*
...
+/filebeat/module/haproxy @elastic/obs-infraobs-integrations
 /filebeat/module/icinga @elastic/integrations
-/filebeat/module/iis @elastic/integrations
-/filebeat/module/kafka @elastic/integrations
+/filebeat/module/iis @elastic/obs-infraobs-integrations
+/filebeat/module/kafka @elastic/obs-infraobs-integrations
 /filebeat/module/kibana @elastic/integrations
 /filebeat/module/kibana/ @elastic/infra-monitoring-ui
 /filebeat/module/logstash @elastic/integrations
 /filebeat/module/logstash/ @elastic/infra-monitoring-ui
```
Comment (joshdover) [inline .github/CODEOWNERS:37]: What's the reason for these two being different? Should they both be infra-monitoring-ui? Same above for Kibana

Comment (andresrc) [reply .github/CODEOWNERS:37]: probably we should remove the one without the final slash

Comment (lalit-satapathy) [reply .github/CODEOWNERS]: updated two entries as part of commit: https://github.com/elastic/beats/pull/35545/commits/cc966fc978300158b35120a7cbd7b1ba9a8a7810

**Comment by lalit-satapathy** [comment]
> @lalit-satapathy : While updating ownerships in Integrations, I realized that below integrations are under ownership of IM-Integrations. We should update the same for module as well ?
> 
> 1. etcd
> 2. coredns

etcd is already added and coredns is platforms.

File: .github/CODEOWNERS (hunk 3 above)
Comment (lalit-satapathy) [reply .github/CODEOWNERS:164]: We will file separate PRs for the cloud filebeat/metricbeat changes.

File: .github/CODEOWNERS (hunk 2 above)
Comment (lalit-satapathy) [reply .github/CODEOWNERS:46]: This can be taken in separate PR later once confirmed

**Comment by lalit-satapathy** [comment]
The plan is to merge this large global changes first and take individual changes separately in future incremental PRs (example: https://github.com/elastic/beats/pull/35545#discussion_r1212586192). We would like to avoid clubbing all changes to single PR here.

 @joshdover, Can we have have someone approve from agent team?

**Review by ishleenk17 — COMMENTED** [review]
Looks good!

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Comment by lalit-satapathy** [comment]
Ping @elastic/elastic-agent-data-plane for code owner approval.

**Review by pierrehilbert — APPROVED** [review]

**Pull request merged**